### PR TITLE
Added door functions to E2.

### PR DESCRIPTION
**Pull request by noatblok**
Added E2 functions to check if the doors are open, and to toggle the doors.

### CODE REVIEW

File: lua/entities/gmod_tardis/modules/sh_doors.lua [hunk 1]
```diff
@@ -83,6 +83,14 @@ if SERVER then
         end
     end
 
+    ENT:AddHook("HandleE2", "door", function(self,name,e2)
+        if name == "Door" then
+            return self:ToggleDoor()
+        elseif name == "GetDoor" and TARDIS:CheckPP(e2.player, self) then
```
Comment (AmyJeanes) [inline lua/entities/gmod_tardis/modules/sh_doors.lua]: We need to move the CheckPP call up to the ToggleDoor if statement rather than this one

**Review by AmyJeanes — CHANGES_REQUESTED** [review]
Hey, thanks for contributing! Just a small little fix needed I think and then it should be good to go 🙂

File: lua/entities/gmod_tardis/modules/sh_doors.lua (hunk 1 above)
Comment (noatblok) [reply lua/entities/gmod_tardis/modules/sh_doors.lua]: Should be fixed.

**Comment by noatblok** [comment]
Updated the E2 description, aswell as having tardisDoor respond with a 1 or a 0 depending on the outcome.

**Review by AmyJeanes — APPROVED** [review]
Looks good!

I've made a small tweak to use the real door state instead of the estimated one which for the purposes of E2 usage makes more sense so you can have a script like this:

```lua
local Success=TARDIS:tardisDoor()
if (Success==1){
    if(TARDIS:tardisDoorOpen()){
        print("TARDIS door open")
    }else{
        print("TARDIS door closed")
    }
}
```

Otherwise it would say the door is open even if it is closing

**Pull request merged**